flang/test/Driver/B-opt.f90

```diff
@@ -0,0 +1,23 @@
+! Check -B driver option.
+!
+! Target triple prefix is not detected for -B.
+! RUN: %flang %s -### -o %t.o -target i386-unknown-linux \
```

Comment:
Use `--target=`

`-target ` has been deprecated since around clang 3.4

Reply:
Thanks. Done. Modified tests for both drivers.

Reply:
@MaskRay Is it OK to land this?